lightning/src/events/mod.rs

```diff
@@ -1114,6 +1114,65 @@ pub enum Event {
 		/// [`ChannelManager`]: crate::ln::channelmanager::ChannelManager
 		channel_type: ChannelTypeFeatures,
 	},
+	/// Indicates a request to open a new dual-funded channel by a peer.
+	///
+	/// To accept the request without contributing funds, call [`ChannelManager::accept_inbound_channel`].
+	/// To accept the request and contribute funds, call [`ChannelManager::accept_inbound_channel_with_contribution`].
+	/// To reject the request, call [`ChannelManager::force_close_without_broadcasting_txn`].
+	///
+	/// The event is always triggered when a new open channel request is received for a dual-funded
+	/// channel, regardless of the value of the [`UserConfig::manually_accept_inbound_channels`]
+	/// config flag. This is so that funding inputs can be manually provided to contribute to the
+	/// overall channel capacity on the acceptor side.
+	///
+	/// [`ChannelManager::accept_inbound_channel`]: crate::ln::channelmanager::ChannelManager::accept_inbound_channel
+	/// [`ChannelManager::accept_inbound_channel_with_contribution`]: crate::ln::channelmanager::ChannelManager::accept_inbound_channel_with_contribution
+	/// [`ChannelManager::force_close_without_broadcasting_txn`]: crate::ln::channelmanager::ChannelManager::force_close_without_broadcasting_txn
+	/// [`UserConfig::manually_accept_inbound_channels`]: crate::util::config::UserConfig::manually_accept_inbound_channels
+	#[cfg(dual_funding)]
```

Comment:
I suggest adding the splicing flag here as well; while this is not strictly needed for splicing, normally splicing is applied on V2 channels (at least as for now), and also, passing two config options to build is not straightforward. It is a simpler model if splicing cfg includes all dual_funding code (but not the other way around).